test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -391,6 +409,119 @@ public void updateRetentionPolicyForCBRTest() throws Exception {
+        String stream = "multiControllerStream" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+        String readerGroupName = "testmultiControllerReaderGroup" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+        // scale to three controller instances.
+        scaleAndUpdateControllerURI(3);
+        // scale to two segment store instances.
+        Futures.getAndHandleExceptions(segmentStoreService.scaleService(2), ExecutionException::new);
+        log.info("Successfully statred 2 instance of segment store service");
+        assertTrue("Creating scope", streamManager.createScope(scope));
+        assertTrue("Creating stream", streamManager.createStream(scope, stream, STREAM_CONFIGURATION));
+
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(scope, controller, connectionFactory);
+
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(stream, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        // Write three event.
+        writingEventsToStream(3, writer, scope, stream);
+
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
+        ReaderGroupConfig readerGroupConfig = getReaderGroupConfig(scope, stream, ReaderGroupConfig.StreamDataRetention.AUTOMATIC_RELEASE_AT_LAST_CHECKPOINT);
+
+        assertTrue("Reader group is not created", readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig));
+        assertEquals(1, controller.listSubscribers(scope, stream).join().size());
+
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
+        AtomicLong clock = new AtomicLong();
+
+        EventStreamReader<String> reader = clientFactory.createReader(readerGroupName + "-" + 1,
+                readerGroupName, new JavaSerializer<>(), readerConfig, clock::get, clock::get);
+        // Read one event with reader.
+        readingEventsFromStream(1, reader);
+
+        CompletableFuture<Checkpoint> checkpoint = initiateCheckPoint("Checkpoint", readerGroup, reader, clock);
+        EventRead<String> read = reader.readNextEvent(READ_TIMEOUT);
+        log.info("Reading next event after checkpoint {}", read.getEvent());
+        Checkpoint cpResult = checkpoint.join();
+        assertTrue(checkpoint.isDone());
+
+        // Write two more events.
+        writingEventsToStream(2, writer, scope, stream);
+
+        // Retention set has one stream cut at 0/150
+        // READER_GROUP_1 updated stream cut at 0/30
+        // Subscriber lower bound is 0/30, truncation should happen at this point
+        // The timeout is set to 2 minutes a little longer than the retention period which is set to 1 minutes
+        // in order to confirm that the retention has taken place.
```

Comment:
Is there a way we can make this faster than 1 minute? 
Retention check frequency comes up in a bunch of tests. Maybe we could detect (Say with env variable or parameter to the controllerFactory) that we are in a test and lower the interval.

Reply:
Currently we have the retention frequency which gets loaded during the controller startup itself as part of the ControllerServiceConfigImpl.java and the value of the retentionFrequency is set to Duration.ofMinutes(Config.MINIMUM_RETENTION_FREQUENCY_IN_MINUTES). So, the minimum value for it that we can give is 1 minute.

Reply:
Raised an issue to track this : https://github.com/pravega/pravega/issues/7037